lib/boom_notifier/error_storage.ex

```diff
@@ -81,53 +75,59 @@ defmodule BoomNotifier.ErrorStorage do
   """
   @spec send_notification?(ErrorInfo.t()) :: boolean()
   def send_notification?(error_info) do
-    %{key: error_hash_key} = error_info
-
-    error_storage_item =
-      Agent.get(:boom_notifier, fn state -> state end)
-      |> Map.get(error_hash_key)
-
-    do_send_notification?(error_storage_item)
+    error_info
+    |> get_error_stats()
+    |> do_send_notification?()
   end
 
   @doc """
   Reset the accumulated_occurrences for the given error info to zero. It also
   increments the max storage capacity based on the notification strategy.
+
+  Returns error storage entry before reset
   """
-  @spec reset_accumulated_errors(error_strategy, ErrorInfo.t()) :: :ok
-  def reset_accumulated_errors(:exponential, error_info) do
-    %{key: error_hash_key} = error_info
+  @spec reset(ErrorInfo.t()) :: __MODULE__.t()
+  @spec reset(ErrorInfo.t(), count_strategy :: :exponential | :always | nil) :: __MODULE__.t()
+  def reset(error_info), do: reset(error_info, nil)
 
-    Agent.update(
-      :boom_notifier,
-      &Map.update!(&1, error_hash_key, fn error_storage_item ->
-        clear_values(error_storage_item)
-        |> Map.update!(:__max_storage_capacity__, fn current -> current * 2 end)
-      end)
-    )
+  def reset(error_info, :exponential) do
+    reset_state(error_info, fn value -> value * 2 end)
   end
 
-  def reset_accumulated_errors([exponential: [limit: limit]], error_info) do
-    %{key: error_hash_key} = error_info
+  def reset(error_info, exponential: [limit: limit]) do
+    reset_state(error_info, fn value -> min(value * 2, limit) end)
+  end
 
-    Agent.update(
-      :boom_notifier,
-      &Map.update!(&1, error_hash_key, fn error_storage_item ->
-        clear_values(error_storage_item)
-        |> Map.update!(:__max_storage_capacity__, fn current -> min(current * 2, limit) end)
-      end)
-    )
+  def reset(error_info, strategy) when strategy in [:always, nil] do
```

Comment:
Maybe we can remove this guard so it always fallbacks here if the strategy doesn't need to update the `max_storage` value